src/ComplexMarkdownParser.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Spatie\YamlFrontMatter;
+
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * A parser that can handle Markdown that contains Markdown.
+ *
+ * Attempts to follow the practices defined in https://jekyllrb.com/docs/front-matter/.
+ */
+class ComplexMarkdownParser
+{
+    /**
+     * The Markdown content.
+     * @var string
+     */
+    protected $content;
+
+    /**
+     * The document string as an array of lines,
+     * making it easier to understand and work with.
+     * @var array
+     */
+    protected $lines;
+
+    /**
+     * The line number of the starting front matter control block.
+     * @var int|null
+     */
+    public $frontMatterStartLine;
+
+    /**
+     * The line number of the ending front matter control block.
+     * @var int|null
+     */
+    public $frontMatterEndLine;
+
+    /**
+     * Construct the parser.
+     * @param string $content The Markdown content to parse.
+     */
+    public function __construct(string $content)
+    {
+        $this->content = $content;
+        $this->lines = explode("\n", $this->content);
```

Comment:
Use single quotes by default.

Reply:
Thank you for the reminder. Here (as far as my understanding of PHP strings goes) we need the double quotes so that the newline character `\n` gets evaluated. With the double quotes, the parser works for inputs using both `LF` and `CRLF `documents. Is there a better way to do this?

I wrote two new tests, to test LF/CRLF line endings and tried switching to using `PHP_EOL`, but the tests failed as my Windows environment could then not parse the LF sequence. I could write a helper method to detect the actual file endings used in each document, but I'm not sure if that solves the issue or just adds more complexity.